src/ipc/ipc4/helper.c

```diff
@@ -590,6 +590,43 @@ static int construct_config(struct ipc4_copier_module_cfg *copier_cfg, uint32_t
 	return IPC4_SUCCESS;
 }
 
+#include "sof/mem_win_debug.h"
+int ipc4_create_io_driver_for_chain(struct ipc4_chain_dma *cdma)
+{
+	struct sof_uuid uuid = {0x6a0a274f, 0x27cc, 0x4afb, {0xa3, 0xe7, 0x34,
+			0x44, 0x72, 0x3f, 0x43, 0x2e}};
+
+	const struct comp_driver *drv;
+	struct comp_dev *dev;
+	uint32_t dir, host_chan, link_chan;
+	int ret;
+
+	drv = ipc4_get_drv((uint8_t *)&uuid);
+	MEM_WIN_DEBUG0(0xbbb0,drv,0,0);
+	if (!drv)
+		return IPC4_INVALID_RESOURCE_ID;
+
+	ret = process_dma_index(cdma->primary.r.host_dma_id, &dir, &host_chan);
+	if (ret != IPC4_SUCCESS)
+		return ret;
+
+	ret = process_dma_index(cdma->primary.r.link_dma_id, &dir, &link_chan);
+	if (ret != IPC4_SUCCESS)
+		return ret;
+
+	dev = drv->ops.chain_dma_create(drv, (uint8_t)host_chan, (uint8_t)link_chan,
```

Comment:
Lets use the API here (which does the NULL check etc).

Reply:
Resolved in:
![image](https://user-images.githubusercontent.com/93590538/203566103-c463b2fe-951f-4b1c-8d40-76aac02c9609.png)